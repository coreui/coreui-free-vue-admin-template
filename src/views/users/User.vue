<template>
  <b-row>
    <b-col cols="12" lg="6">
      <b-card no-header>
        <template slot="header">
          User id:  {{ $route.params.id }}
        </template>
        <b-table striped small fixed responsive="sm" :items="items($route.params.id)" :fields="fields">
          <template slot="value" slot-scope="data">
            <strong>{{data.item.value}}</strong>
          </template>
        </b-table>
        <template slot="footer">
          <b-button @click="goBack">Back</b-button>
        </template>
      </b-card>
    </b-col>
  </b-row>
</template>

<script>
import usersData from './UsersData'
export default {
  name: 'User',
  props: {
    caption: {
      type: String,
      default: 'User id'
    },
  },
  data: () => {
    return {
      items: (id) => {
        const user = usersData.find( user => user.id.toString() === id)
        const userDetails = user ? Object.entries(user) : [['id', 'Not found']]
        return userDetails.map(([key, value]) => {return {key: key, value: value}})
      },
      fields: [
        {key: 'key'},
        {key: 'value'},
      ],
    }
  },
  methods: {
    goBack() {
      this.$router.go(-1)
      // this.$router.replace({path: '/users'})
    }
  }
}
</script>
