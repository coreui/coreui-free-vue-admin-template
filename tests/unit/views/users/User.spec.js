import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import CoreuiVue from '@coreui/vue'
import User from '@/views/users/User'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('User.vue', () => {
  it('has a name', () => {
    expect(User.name).toBe('User')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(User, {
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is User', () => {
    const wrapper = shallowMount(User, {
      localVue,
      router
    })
    expect(wrapper.is(User)).toBe(true)
  })
  it('should have methods', () => {
    const wrapper = shallowMount(User,{
      localVue,
      router
    })

    expect(typeof User.methods.goBack ).toEqual('function')
    expect(wrapper.vm.goBack()).toBeUndefined()
    expect(typeof User.methods.getUserData ).toEqual('function')
    expect(wrapper.vm.getUserData(1)).toStrictEqual([
      { key: 'username', value: 'Samppa Nori' },
      { key: 'registered', value: '2012/01/01' },
      { key: 'role', value: 'Member' },
      { key: 'status', value: 'Active' } ])
    expect(wrapper.vm.getUserData(30)).toStrictEqual([{"key": "id", "value": "Not found"}])
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(User, {
      localVue,
        router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
