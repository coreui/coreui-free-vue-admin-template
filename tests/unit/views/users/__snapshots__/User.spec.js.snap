// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.vue renders correctly 1`] = `
<crow-stub
  gutters="true"
  tag="div"
>
  <ccol-stub
    col="12"
    lg="6"
    tag="div"
  >
    <ccard-stub>
      <ccardheader-stub>
        
        User id:  
      
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cdatatable-stub
          fields="[object Object],[object Object]"
          fixed="true"
          header="true"
          items="[object Object]"
          itemsperpage="10"
          responsive="true"
          small=""
          sortervalue="[object Object]"
          striped="true"
        />
      </ccardbody-stub>
       
      <ccardfooter-stub>
        <cbutton-stub
          activeclass="active"
          color="primary"
          event="click"
          exactactiveclass="active"
          routertag="a"
          target="_self"
          type="button"
        >
          Back
        </cbutton-stub>
      </ccardfooter-stub>
    </ccard-stub>
  </ccol-stub>
</crow-stub>
`;
